go/chat/server.go

```diff
@@ -599,21 +599,18 @@ func (h *Server) NewConversationLocal(ctx context.Context, arg chat1.NewConversa
 		switch arg.MembersType {
 		case chat1.ConversationMembersType_TEAM, chat1.ConversationMembersType_IMPTEAM:
 			joinMessageBody := chat1.NewMessageBodyWithJoin(chat1.MessageJoin{})
-			rl, err = h.postJoinLeave(ctx, uid.ToBytes(), convID, joinMessageBody)
+			irl, err := postJoinLeave(ctx, h.G(), h.remoteClient, uid.ToBytes(), convID, joinMessageBody)
 			if err != nil {
 				h.Debug(ctx, "posting join-conv message failed: %v", err)
 				// ignore the error
 			}
-			if err == nil && rl != nil {
-				res.RateLimits = append(res.RateLimits, *rl)
-			}
+			res.RateLimits = append(res.RateLimits, irl...)
```

Comment:
Using return values from functions that have errored seems like asking for trouble. But I guess if you want to keep it here it's probably the least sketchy case of that pattern.